src/ipc/mem/PageStack.cc

```diff
@@ -15,128 +15,133 @@
 #include "ipc/mem/Page.h"
 #include "ipc/mem/PageStack.h"
 
-/// used to mark a stack slot available for storing free page offsets
-const Ipc::Mem::PageStack::Value Writable = 0;
-
-Ipc::Mem::PageStack::PageStack(const uint32_t aPoolId, const unsigned int aCapacity, const size_t aPageSize):
-    thePoolId(aPoolId), theCapacity(aCapacity), thePageSize(aPageSize),
-    theSize(theCapacity),
-    theLastReadable(prev(theSize)), theFirstWritable(next(theLastReadable)),
-    theItems(aCapacity)
+/* Ipc::Mem::PageStackStorageSlot */
+
+// We are using uint32_t for Pointer because PageId::number is uint32_t.
```

Comment:
This check does not care about what the type size is, so long as they are matching.

IMO this text should be at the place the Pointer type is defined. It may also be useful there to indicate "larger" means over 16TB or whatever N*4 GB is for page size N.

Reply:
Done.